src/e3/os/platform.py

```diff
@@ -44,10 +44,10 @@ class SystemInfo:
     ld_info = None
 
     # Cache for SystemInfo methods
-    _platform = None
-    _os_version = None
-    _is_virtual = None
-    _hostname = None
+    _platform: Optional[str] = None
+    _os_version: Optional[Tuple[str, str]] = None
+    _is_virtual: Optional[bool] = None
+    _hostname: Optional[Tuple[str, str]] = None
```

Comment:
FWIW, it would have been nice to take this opportunity to document those class variables as well, particularly the ones that are tuples, because it's less obvious (at least to me) why we need tuples.